### PR TITLE
Admin holiday 

### PR DESCRIPTION
**Pull request by Namrata12gupta**

### CODE REVIEW

File: spec/controllers/admin_main/holiday_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  describe '#create' do
+    it 'response successfully when holiday created' do
+      post :create, params: { holiday: { holiday_name: 'xyz', holiday_date: '2022-04-23', holiday_type: :National } }
+      expect(response.status).to eq(302)
+    end
+  end
+
+  describe '#edit' do
+    it 'response successfully when holiday exit' do
```
Comment (rorpyll) [inline spec/controllers/admin_main/holiday_controller_spec.rb:35]: write spec for not found condition

File: spec/controllers/admin_main/holiday_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      get :index
+      expect(assigns(:holidays).count).to eq(2)
+      expect(response.status).to eq(200)
+    end
+  end
+
+  describe '#new' do
+    it 'show holiday form' do
+      get :new
+      expect(response.status).to eq(200)
```
Comment (rorpyll) [inline spec/controllers/admin_main/holiday_controller_spec.rb:23]: write spec for not found condition

File: spec/controllers/admin_main/holiday_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    end
+  end
+
+  describe '#create' do
```
Comment (rorpyll) [inline spec/controllers/admin_main/holiday_controller_spec.rb:27]: write spec for not create condition

**Review by rorpyll — CHANGES_REQUESTED** [review]

File: spec/controllers/admin_main/holiday_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  end
+
+  describe '#edit' do
+    it 'response successfully when holiday exit' do
+      get :edit, params: { id: holiday.id }
+      expect(assigns(:holiday)).to eq(holiday)
+    end
+  end
+
+  describe '#update' do
```
Comment (ror-uday) [inline spec/controllers/admin_main/holiday_controller_spec.rb:41]: write spec for when holiday not update

**Review by ror-uday — CHANGES_REQUESTED** [review]

File: app/controllers/admin_main/holidays_controller.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+      if @holiday.update(holiday_params)
+        redirect_to admin_main_holidays_path
+      else
+        render :edit, status: :unprocessable_entity
+      end
+    end
+
+    def destroy
+      @holiday = Holiday.find(params[:id])
```
[the remark this answers is not in the document]
Comment (Namrata12gupta) [reply app/controllers/admin_main/holidays_controller.rb:37]: changes done

File: app/views/admin_main/holidays/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,31 @@
+</aside><!-- End Sidebar-->
+
+  <main id="main" class="main">
+    <h1>Editing Holiday</h1>
+    <h1 style="width: 300px;"></h1>
+
+    <%= form_with model: @holiday, :url => admin_main_holiday_path, :method => "patch" do |form| %>
```
[the remark this answers is not in the document]
Comment (Namrata12gupta) [reply app/views/admin_main/holidays/edit.html.erb:7]: changes done

File: app/views/admin_main/holidays/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+</aside><!-- End Sidebar-->
+
+  <main id="main" class="main">
```
[the remark this answers is not in the document]
Comment (Namrata12gupta) [reply app/views/admin_main/holidays/new.html.erb:3]: changes done

File: app/views/shared/_header.html.erb [hunk 8]
```diff
@@ -123,18 +131,35 @@
...
+          <% else %> 
+          <%= link_to  admin_main_holidays_path , class: "nav-link collapsed" do %>
+          <i class="bi bi-menu-button-wide"></i>
+          <%= 'Holidays' %>
+          <% end %>
+          <% end %>
       </li>
+
       <li class="nav-item">
+         <% if current_employee.role.name == "Employee" || current_employee.role.name == "HR" %>
```
[the remark this answers is not in the document]
Comment (Namrata12gupta) [reply app/views/shared/_header.html.erb:150]: changes done

File: spec/controllers/admin_main/holiday_controller_spec.rb (hunk 1 above)
Comment (Namrata12gupta) [reply spec/controllers/admin_main/holiday_controller_spec.rb:35]: changes done

File: spec/controllers/admin_main/holiday_controller_spec.rb (hunk 4 above)
Comment (Namrata12gupta) [reply spec/controllers/admin_main/holiday_controller_spec.rb:41]: changes done

File: app/models/ability.rb [hunk 9]
```diff
@@ -6,7 +6,7 @@ class Ability
   def initialize(employee)
     # Define abilities for the passed in user here. For example:
     #   user ||= User.new # guest user (not logged in)
-        if employee.role.name == "HR"
+        if employee.role.name == "HR" || employee.role.name == "Admin"
```
Comment (ror-uday) [inline app/models/ability.rb:9]: use instance method

**Review by ror-uday — CHANGES_REQUESTED** [review]

**Review by ror-uday — APPROVED** [review]

File: app/views/shared/_header.html.erb [hunk 10]
```diff
@@ -1,151 +1,159 @@
...
+    </ul>
+  </nav><!-- End Icons Navigation -->
+</header><!-- End Header -->
 
   <!-- ======= Sidebar ======= -->
   <aside id="sidebar" class="sidebar">
-
     <ul class="sidebar-nav" id="sidebar-nav">
-
+
```
Comment (chiraggargsass) [inline app/views/shared/_header.html.erb:52]: remove blank space

File: app/views/shared/_header.html.erb [hunk 11]
```diff
@@ -1,151 +1,159 @@
...
-          <% if current_employee.role.name == "HR" %>
-          <%= link_to profile_path , class: "nav-link collapsed" do %>
-          <i class="bi bi-person"></i>
-              <%= 'Profile' %>
-          <% end %>
-          <% end %>
+        <% if current_employee.is_hr? %>
+        <%= link_to profile_path , class: "nav-link collapsed" do %>
+        <i class="bi bi-person"></i>
+            <%= 'Profile' %>
```
Comment (chiraggargsass) [inline app/views/shared/_header.html.erb:64]: remove ruby tag

File: app/views/shared/_header.html.erb [hunk 12]
```diff
@@ -1,151 +1,159 @@
...
+        <% end %>
+        <% end %>
+      </li>
+
+      <li class="nav-item">
+        <% if current_employee.is_admin? %>
```
Comment (chiraggargsass) [inline app/views/shared/_header.html.erb:70]: should be use OR(||) condition for HR and ADMIN.
not to use same code in both condition.

File: app/views/shared/_header.html.erb [hunk 13]
```diff
@@ -1,151 +1,159 @@
...
-          <% unless current_employee.role.name == "HR" %>
-              <%= 'Profile' %>
-          <% else %>
-              <%= 'Employees' %>
-          <% end %>
-          <% end %>
+        <%= link_to employees_path , class: "nav-link collapsed" do %>
+        <i class="bi bi-person"></i>
+        <% unless current_employee.is_hr? ||current_employee.is_admin?%>
+            <%= 'Profile' %>
```
Comment (chiraggargsass) [inline app/views/shared/_header.html.erb:82]: Remove Ruby tag

File: config/routes.rb [hunk 14]
```diff
@@ -35,10 +35,18 @@
...
+
     resources :attendences
     get '/emp-attendance/:id', to: 'attendences#show_attendence', as: 'show_attendance'
     get '/search', to: 'attendences#search'
   end
+
+  namespace :admin_main do
+    resources :employees 
+      resources :holidays
+        resources :daily_tasks
```
Comment (chiraggargsass) [inline config/routes.rb:47]: add proper indentation

**Review by chiraggargsass — CHANGES_REQUESTED** [review]

File: app/views/shared/_header.html.erb (hunk 10 above)
Comment (Namrata12gupta) [reply app/views/shared/_header.html.erb:52]: done

File: app/views/shared/_header.html.erb (hunk 11 above)
Comment (Namrata12gupta) [reply app/views/shared/_header.html.erb:64]: done

File: app/views/shared/_header.html.erb (hunk 12 above)
Comment (Namrata12gupta) [reply app/views/shared/_header.html.erb:70]: done

File: config/routes.rb (hunk 14 above)
Comment (Namrata12gupta) [reply config/routes.rb:47]: done

File: app/views/shared/_header.html.erb (hunk 13 above)
Comment (Namrata12gupta) [reply app/views/shared/_header.html.erb:82]: done

File: app/controllers/admin_main/holidays_controller.rb [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+      else
+        render :index
+      end
+    end
+
+    private
+
+    def set_holiday
+      @holiday = Holiday.find_by_id params[:id]
+      redirect_to root_path, { notic: 'NOT FOUND' } unless @holiday.present?
```
Comment (ror-uday) [inline app/controllers/admin_main/holidays_controller.rb:48]: spelling mistake notice

**Review by ror-uday — CHANGES_REQUESTED** [review]

File: app/controllers/admin_main/holidays_controller.rb (hunk 15 above)
Comment (Namrata12gupta) [reply app/controllers/admin_main/holidays_controller.rb:48]: done

**Review by chiraggargsass — APPROVED** [review]

**Pull request merged**